### PR TITLE
send tipset identifier to node when interacting with chain

### PR DESCRIPTION
**Pull request by laser**
## Why does this PR exist?

This is the last (hopefully) PR in a chain of PRs which pass chain state identifiers to methods on the `Node*` interfaces.

### CODE REVIEW

File: storagemarket/testnodes/testnodes.go [hunk 1]
```diff
@@ -180,8 +180,8 @@ func (n *FakeClientNode) OnDealSectorCommitted(ctx context.Context, provider add
 }
 
 // ValidateAskSignature returns the stubbed validation error
```
Comment (shannonwells) [inline storagemarket/testnodes/testnodes.go]: fix comment

File: storagemarket/types.go [hunk 2]
```diff
@@ -321,7 +321,7 @@ type StorageProviderNode interface {
 
 	OnDealSectorCommitted(ctx context.Context, provider address.Address, dealID abi.DealID, cb DealSectorCommittedCallback) error
 
-	LocatePieceForDealWithinSector(ctx context.Context, dealID abi.DealID) (sectorID uint64, offset uint64, length uint64, err error)
+	LocatePieceForDealWithinSector(ctx context.Context, dealID abi.DealID, tok shared.TipSetToken) (sectorID uint64, offset uint64, length uint64, err error)
```
Comment (hannahhoward) [inline storagemarket/types.go]: Is this method actually talking to the chain? I believe it's not.. though I could be wrong.

Comment (laser) [reply storagemarket/types.go]: Yep. The go-filecoin implementation of this method requires both storage miner actor and storage miner actor states.

**Comment by laser** [comment]
@hannahhoward 

> For the forward facing APIs on Client and Provider, I would prefer not to augment them with a tipset token, and instead have them internally call GetMoreRecentStateID internally

That makes sense to me. I'll make that update tomorrow.

**Comment by laser** [comment]
> Also, should LocatePieceForDealWithinSector take a tipset token? My understanding is this doesn't really talk to the chain so much as it uses the storage miner code directly. This is non-blocking if you can confirm my understanding is wrong.

Yep, [indeed it does](https://github.com/filecoin-project/go-filecoin/blob/7748bab472a148decb704178b03cde6516a305dd/internal/app/go-filecoin/connectors/storage_market/provider.go#L140). At least in go-filecoin.

File: storagemarket/testnodes/testnodes.go (hunk 1 above)
Comment (laser) [reply storagemarket/testnodes/testnodes.go]: done!

**Pull request merged**